src/if_perl.xs

```diff
@@ -56,6 +56,8 @@
 #ifdef __GNUC__
 # pragma GCC diagnostic push
 # pragma GCC diagnostic ignored "-Wunused-variable"
+# pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
+# pragma GCC diagnostic ignored "-Wshadow"
```

Comment:
I think should not ignore these warnings (`-Wincompatible-pointer-types` and `-Wshadow`) by pragma.
For CI, should add them to `PERL_CFLAGS_EXTRA` in `ci/config.mk.sed`.

Reply:
done: https://github.com/vim/vim/pull/12575/commits/6abf8cdab0bf03ea05cec19d538d4387d730a933